server/routes/auth.js

```diff
@@ -50,133 +53,140 @@ router.get('/register', async (req, res) => {
+                                error: true,
+                                message: 'somthing went wrong'
+                            })
+                        }
+
+                        if (result) {
+
+                            const token = jwt.sign({ _id: user._id }, process.env.JWT_SECRET, { expiresIn: '1h' })
+                            req.session.authorization = token
+                            return res.send({
+                                success: true,
+                                isAuth: 'Login successfully',
+                                token
+                            })
+                        } else {
+                            return res.send({
+                                error: true,
+                                message: 'email or password is incorrect'
+                            });
+                        }
+                    })
+                } else {
+                    res.send({
+                        error: true,
+                        message: "email or password is incorrect"
+                    })
+                }
+            }).catch((error) => {
+                console.error(error)
+            })
+    } catch (error) {
+
+    }
+});
+
+router.post('/logout', (req, res) => {
+
+    req.session.destroy((error) => {
+        if (!error) {
+            res.send({
+                success: true,
+                message: "Logged out."
+            })
+        }
+        if (error) {
+            res.send({
+                error: true,
+                message: 'an error has been encountered'
+            })
+        };
+    })
+})
+
+router.post('/post', async (req, res) => {
+    try {
+        const {
+            title,
+            content
+        } = req.body
+
+        const token = req.session.authorization;
+        if (token) {
+            jwt.verify(token, process.env.JWT_SECRET, async (err, decodeToken) => {
+                if (err) {
+                    res.locals.user = null;
+                } else {
+                    await User.findById(decodeToken._id).then(user => {
+                        post = new postSchema({
+                            title,
+                            content,
+                            author: user.author,
+                            avatar: user.avatar,
+                            date: new Date()
+                        })
+                        post.save()
+                        res.send({
+                            success: true,
+                            message: 'post has been successfully saved'
+                        })
+                    })
+                }
+            })
+        }
+    } catch (error) {
+        res.send({
+            error: true,
+            message: error,
+        })
+    }
+})
+
+router.get('/post', async (req, res) => {
+    postSchema.find()
+        .then(post => {
+            res.send(post)
+        })
+})
+
+router.get('/logged', async (req, res) => {
```

Comment:
Usually it's called /profile and it should return the whole profile object except for password